src/init/features/apphosting/index.ts

```diff
@@ -66,27 +72,50 @@ export async function doSetup(setup: any, projectId: string): Promise<void> {
     }
     logWarning(`Backend with id ${backendId} already exists in ${location}`);
   }
-  const backend: Backend = await onboardBackend(projectId, location, backendId);
+  const backend = await onboardBackend(projectId, location, backendId);
+
+  const branch = await promptOnce({
+    name: "branch",
+    type: "input",
+    default: "main",
+    message: "Which branch do you want to deploy?",
+  });
+  const build = await createBuild(projectId, location, backendId, {
+    source: {
+      codebase: {
+        branch,
+      },
+    },
+  });
+
+  if (build.state !== "READY") {
+    throw new FirebaseError(
+      `Failed to build your app. Please inspect the build logs at ${build.buildLogsUri}.`,
+      { children: [build.error] }
+    );
+  }
+
+  await setDefaultTraffic(projectId, location, backendId, build.name.split("/").pop()!);
 
   if (backend) {
     logSuccess(`Successfully created backend:\n\t${backend.name}`);
-    logSuccess(`Your site is being deployed at:\n\thttps://${backend.uri}`);
+    logSuccess(`Your site is now deployed at:\n\thttps://${backend.uri}`);
```

Comment:
why change?

Reply:
Becaused at this point, the site would've have been deployed!